test/integration/extnet_test.go

```diff
@@ -0,0 +1,258 @@
+	case "crio":
+	    t.Skip("skipping: access direct test is broken on windows: https://github.com/kubernetes/minikube/issues/8304")
+	case "docker":
+	}
+
+	extnetNetworkName := fmt.Sprintf("%s-%s", "network-extnet", fmt.Sprintf("%06d", time.Now().UnixNano()%1000000))
+
+	// Serial tests
+	t.Run("serial", func(t *testing.T) {
+		tests := []struct {
+			name      string
+			validator validateFunc
+		}{
+			{"CreateExtnet", createExtnet},
+			{"FreshStart", extnetValidateFreshStart},
+			{"ConnectExtnet", connectExtnet},
+			{"Stop", extnetValidateStop},
+			{"VerifyStatus", extnetValidateStatus},
+			{"Start", extnetValidateStart},
+			{"Delete", extnetValidateDelete},
+		//	{"Fail", fail},
+			{"VerifyDeletedResources", extnetValidateVerifyDeleted},
+		}
+		for _, tc := range tests {
+			tc := tc
+
+			if ctx.Err() == context.DeadlineExceeded {
+				t.Fatalf("Unable to run more tests (deadline exceeded)")
+			}
+			if t.Failed() {
+				// t.Fatalf("Previous test failed, not running dependent tests")
+				break
+			}
+
+			t.Run(tc.name, func(t *testing.T) {
+				tc.validator(ctx, t, profile, extnetNetworkName)
+				if t.Failed() && *postMortemLogs {
+					PostMortemLogs(t, profile)
+				}
+			})
+		}
+
+		t.Run("DeleteExtnet", func(t *testing.T) {
+			deleteExtnet(ctx, t, profile, extnetNetworkName)
+			if t.Failed() && *postMortemLogs {
+				PostMortemLogs(t, profile)
+			}
+		})
+	})
+}
+
+// connectExtnet create a docker network
+func createExtnet(ctx context.Context, t *testing.T, profile string, extnetNetworkName string) {
+	subnet := "172.28.0.0/16"
+	ipRange :="172.28.0.0/24"
+	cmd := exec.CommandContext(ctx, "docker", "network", "create", extnetNetworkName, fmt.Sprintf("--subnet=%s", subnet), fmt.Sprintf("--ip-range=%s", ipRange))
+
+	result, err := Run(t, cmd)
+	if err != nil {
+		t.Fatalf("failed to execute 'docker network create', error: %v, output: %s", err, result.Output())
+	}
+	extnetNetworkID := result.Output()
+	fmt.Fprintf(os.Stderr, "%s", extnetNetworkID)
+}
+
+// connectExtnet connect network to the minikube cluster
+func connectExtnet(ctx context.Context, t *testing.T, profile string, extnetNetworkName string) {
+	cmd := exec.CommandContext(ctx, "docker", "network", "connect", extnetNetworkName, profile)
+
+	result, err := Run(t, cmd)
+	if err != nil {
+		t.Fatalf("failed to execute 'docker network connect', error: %v, output: %s", err, result.Output())
+	}
+	extnetNetworkID := result.Output()
+	fmt.Fprintf(os.Stderr, "%s", extnetNetworkID)
+}
+
+// deleteExtnet just starts a new minikube cluster
+func deleteExtnet(ctx context.Context, t *testing.T, profile string, extnetNetworkName string) {
+	cmd := exec.CommandContext(ctx, "docker", "network", "rm", extnetNetworkName)
+
+	result, err := Run(t, cmd)
+	if err != nil {
+		t.Fatalf("failed to execute 'docker network delete', error: %v, output: %s", err, result.Output())
+	}
+	extnetNetworkID := result.Output()
+	fmt.Fprintf(os.Stderr, "%s", extnetNetworkID)
+}
```

Comment:
as commented above, this becomes our deferred function called when tests ends (regardless how):

```
func deleteExtnet(t *testing.T, extNetworkName string) {
	ctx, cancel := context.WithTimeout(context.Background(), Seconds(10))
	defer cancel()

	cmd := exec.CommandContext(ctx, "docker", "network", "rm", extNetworkName)

	result, err := Run(t, cmd)
	if err != nil {
		t.Fatalf("failed to execute 'docker network delete', error: %v, output: %s", err, result.Output())
	}
	t.Logf("deleted new network %q", extNetworkName)
}
```

Reply:
I did the defer trick, see below